src/components/createCourse/CreateCourse.tsx

```diff
@@ -0,0 +1,280 @@
+import React, { useState } from 'react';
+import Button from '../../common/Button';
+import './CreateCourse.css';
+import { CreateCourseProps } from './CreateCourse.types';
+import { getCourseDuration } from '../../helpers/getCourseDuration';
+
+const CreateCourse: React.FC<CreateCourseProps> = ({
+  setCoursesList,
+  setAuthorsList,
+  setCreateMode,
+  authorsList,
+}) => {
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [courseAuthors, setCourseAuthors] = useState([]);
+  const [newAuthor, setNewAuthor] = useState('');
+  const [duration, setDuration] = useState(null);
+
+  const [errors, setErrors] = useState({
+    title: false,
+    description: false,
+    duration: false,
+    courseAuthors: false,
+    newAuthor: false,
+  });
+
+  const handleTitle = (event) => {
+    setErrors((prev) => ({
+      ...prev,
+      title: false,
+    }));
+    setTitle(event.target.value);
+  };
+
+  const handleDescription = (event) => {
+    setErrors((prev) => ({
+      ...prev,
+      description: false,
+    }));
+    setDescription(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    let hasErrors = false;
+    if (!title || title.length < 2) {
+      setErrors((prev) => ({
+        ...prev,
+        title: true,
+      }));
+      hasErrors = true;
+    }
+    if (!description || description.length < 2) {
+      setErrors((prev) => ({
+        ...prev,
+        description: true,
+      }));
+      hasErrors = true;
+    }
+
+    if (!duration) {
+      setErrors((prev) => ({
+        ...prev,
+        duration: true,
+      }));
+      hasErrors = true;
+    }
+
+    if (!courseAuthors.length) {
+      setErrors((prev) => ({
+        ...prev,
+        courseAuthors: true,
+      }));
+      hasErrors = true;
+    }
+
+    if (hasErrors) {
```

Comment:
You can check `errors` object to have at least one `true` field instead of maintaining a `hasError`.